### ISSUE TITLE
Rust 2018: `cargo fix` doesn't warn about reserved keyword `try`

### ISSUE THREAD

**Issue by Procrat** [rust-lang/rust#53921]
I had some old code lying around that still used the `try!` macro instead of the `?` operator. `cargo fix` didn't output any errors or warnings before switching to Rust 2018. After switching, `cargo build` resulted in:

```
error: expected expression, found reserved keyword `try`
   --> src/...
    |
183 |         try!(...);
    |         ^^^ expected expression
```

It sounds to me like `rustc`'s behaviour is actually right here, but `cargo fix`'s is not. Should I be filing this with [rustfix](https://github.com/rust-lang-nursery/rustfix) instead? The [edition transition guide](https://rust-lang-nursery.github.io/edition-guide/editions/transitioning-your-code-to-a-new-edition.html) pointed me here though.

**Comment by zackmdavis** [comment]
(How we decide to fix this will likely depend on the outcome of ongoing discussion in #53686.)
